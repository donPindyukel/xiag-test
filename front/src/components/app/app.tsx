import React from "react";
import "./styles.css";
import { CreateQuestion, Poll } from "../pages";
import { Header } from "Elements";
import { BrowserRouter, Route } from "react-router-dom";

export class App extends React.Component<{}> {
  render() {
    return (
      <div className="app-wrap">
        <Header/>
        <BrowserRouter>
          <Route path="/" exact component={CreateQuestion} />
          <Route path="/:pollId" component={Poll} />
        </BrowserRouter>
      </div>
    );
  }
}
