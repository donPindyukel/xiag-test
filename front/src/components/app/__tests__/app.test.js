import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { App } from "../app";

describe("<App />", () => {
  let component;

  describe("App", () => {
    beforeAll(() => {
      component = shallow(<App />);
    });

    it("renders the component", () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
