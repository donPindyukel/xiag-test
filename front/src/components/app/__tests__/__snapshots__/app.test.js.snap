// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App renders the component 1`] = `
<div
  className="app-wrap"
>
  <Component />
  <BrowserRouter>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/:pollId"
    />
  </BrowserRouter>
</div>
`;
